src/TestContext.elm

```diff
@@ -1309,6 +1425,153 @@ advanceTo end testContext =
                                     }
 
 
+{-| Lets you assert on the values that have been sent to an outgoing port.
+
+The parameters are:
+
+1.  The name of the port
+2.  A JSON decoder corresponding to the type of the port
+3.  A function that will receive the list of values sent to the port
+    since the last use of `checkAndClearOutgoingPort` (or since the start of the test)
+    and returns an `Expectation`
+
+For example:
+
+    ...
+        |> checkAndClearOutgoingPort
+            "saveApiTokenToLocalStorage"
+            Json.Decode.string
+            (Expect.equal [ "975774a26612", "920facb1bac0" ])
+        |> ...
+
+NOTE: You must use [`withSimulatedEffects`](#withSimulatedEffects) before you call [`start`](#start) to be able to use this function.
+
+-}
+checkAndClearOutgoingPort : String -> Json.Decode.Decoder a -> (List a -> Expectation) -> TestContext msg model effect -> TestContext msg model effect
```

Comment:
🎨 I think this name makes sense, but I can imagine people searching for `simulateOutgoingPort`.

Reply:
Maybe "simulateAndClearOutgoingPort" ?

Reply:
That seems better to me, especially since it makes it clear it's related to the family of `simulate*` functions.

Reply:
Hmm...   I guess what's weird is that it doesn't actually simulate anything -- it's just checking the simulation state resulting from things that previously were simulated -- basically like `assertHttpRequestWasMade`, etc.

Reply:
Oh, I see. So calling this function doesn't actually affect what `simulateIncomingPort` does (unless of course it causes the whole `TestContext` to fail).

Maybe `assertAndClearOutgoingPort` makes more sense then? Seems like the "clear" is important as well, since you can't call the same assertion twice (like you could with `assertHttpRequest`), since it updates the internal state of the `TestContext`.

Reply:
Renamed to assertAndClearOutgoingPortValues